classes/quiz_settings.php

```diff
@@ -402,4 +438,49 @@ private function split_keys(string $keys) : array {
         }
         return $keys;
     }
+
+    /**
+     * Try and get a file in the user draft filearea by itemid.
+     *
+     * @param string $itemid Item ID of the file.
+     * @return stored_file|null Returns null if no file is found.
+     *
+     * @throws \coding_exception
+     */
+    private function get_current_user_draft_file(string $itemid) : ?stored_file { // @codingStandardsIgnoreLine
+        global $USER;
+        $context = context_user::instance($USER->id);
+        $fs = get_file_storage();
+        $files = $fs->get_area_files($context->id, 'user', 'draft', $itemid);
```

Comment:
In bunch of places in moodle  I saw something like 
```
if (!$files = $fs->get_area_files($context->id, 'user', 'draft', $draftid, 'id DESC', false)) {
                return false;
}
 $file = reset($files);They don't include dirs and change sort order to get the latest file. 
```
Do you think we need to change it here as well?